src/dbnode/x/xio/segment_reader.go

```diff
@@ -46,40 +47,101 @@ func (sr *segmentReader) Clone(
 	return NewSegmentReader(sr.segment.Clone(pool)), nil
 }
 
-func (sr *segmentReader) Read(b []byte) (int, error) {
-	if len(b) == 0 {
-		return 0, nil
+func (sr *segmentReader) Read64() (word uint64, n byte, err error) {
+	sr.lazyInit()
+
+	var (
+		nh    = len(sr.lazyHead)
```

Comment:
nit: the variable names are a little hard to parse here

Reply:
Good point, will rename:
```go
		headLen     = len(sr.lazyHead)
		headTailLen = headLen + len(sr.lazyTail)
```